### ISSUE TITLE
NoUnusedImports false negative for function when property name matches

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#5888]
I would expect the following code to fail with `NoUnusedImports`:

```kotlin
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.focus.FocusRequester
import androidx.compose.ui.focus.focusRequester

@Composable
fun TestComposable() {
  val focusRequester = remember { FocusRequester() }

  LaunchedEffect(Unit) {
    focusRequester.requestFocus()
  }
}
```

but it does not.

If I rename the property to `foo` it does fail:

```kotlin
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.focus.FocusRequester
import androidx.compose.ui.focus.focusRequester

@Composable
fun TestComposable() {
  val foo = remember { FocusRequester() }

  LaunchedEffect(Unit) {
    foo.requestFocus()
  }
}
```

**Comment by atulgpt** [comment]
Hi @cortinico/ @eygraber  `NoUnusedImports` is provided by the `KtLint` so should there be any duplicate issue over there?

**Comment by BraisGabin** [comment]
Then yes, I'm closing this because that's an issue on ktlint. Am issue there should be raised. I think that our own rule shouldn't have that issue as we use type solving.

**Comment by eygraber** [comment]
Ok I'll try with ktlint.

**Comment by eygraber** [comment]
https://github.com/pinterest/ktlint/issues/1915
